packages/core/src/rum/instrumentation/ShallowObjectEqualityChecker.ts

```diff
@@ -0,0 +1,16 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+/**
+ * Does a shallow comparison between 2 objects
+ */
+export const areObjectShallowEqual = (objectA: Record<string, unknown>, objectB: Record<string, unknown>): boolean => {
+    const keys = Object.keys(objectA);
+    for (const key of keys) {
+        if (objectA[key] !== objectB[key]) return false
+    }
+    return keys.length === Object.keys(objectB).length;
```

Comment:
Shouldn't we do the length check first ?

Reply:
In terms of performance I believe it is slightly better to do it at the end as props length very rarely change, but still probably negligable.
I don't see a big difference in terms of comprehension of the code, but if you see one I am happy to move the line above :)